monai/bundle/scripts.py

```diff
@@ -116,10 +117,14 @@ def _pop_args(src: dict, *args: Any, **kwargs: Any) -> tuple:
 
 
 def _log_input_summary(tag: str, args: dict) -> None:
-    logger.info(f"--- input summary of monai.bundle.scripts.{tag} ---")
+    log_buffer = StringIO()
+
+    log_buffer.write(f"--- input summary of monai.bundle.scripts.{tag} ---\n")
     for name, val in args.items():
-        logger.info(f"> {name}: {pprint_edges(val, PPRINT_CONFIG_N)}")
-    logger.info("---\n\n")
+        log_buffer.write(f"> {name}: {pprint_edges(val, PPRINT_CONFIG_N)}\n")
+    log_buffer.write("---\n\n")
+
+    logger.info(log_buffer.getvalue())
```

Comment:
This change may still not solve the issue mentioned in https://github.com/Project-MONAI/MONAI/issues/7513?

Reply:
What it seems to me will happen here is the message will be written all at once with one call to `logger.info()` which has a greater chance of appearing all at once in the output in the presence of multiple GPUs. This is only by chance and isn't guaranteed. What we need in the case of multiple GPUs is to identify which process is rank 0 and only it writes to the logger by default but with an option to allow everyone to write if desired. With the `torch.distributed` package this would be:

```python
import torch.distributed as dist
...
rank = dist.get_rank() if dist.is_initialized() else 0
```

Reply:
@ericspod  Thanks for your advise. I will try it.